File: examples/python/pagination/_async/sandbox.py
import asyncio

from daytona import AsyncDaytona, ListSandboxesQuery, SandboxListSortDirection, SandboxListSortField, SandboxState


async def main():
    async with AsyncDaytona() as daytona:
        async for sandbox in daytona.list(
            ListSandboxesQuery(
                limit=10,
                labels={"env": "dev"},
                states=[SandboxState.STARTED],
                sort=SandboxListSortField.CREATEDAT,
                order=SandboxListSortDirection.DESC,
            )
        ):
            print(sandbox.id)


if __name__ == "__main__":
    asyncio.run(main())

File: examples/typescript/pagination/sandbox.ts
import { Daytona, SandboxListSortDirection, SandboxListSortField, SandboxState } from '@daytona/sdk'

async function main() {
  const daytona = new Daytona()

  for await (const sandbox of daytona.list({
    limit: 10,
    labels: { env: 'dev' },
    states: [SandboxState.STARTED],
    sort: SandboxListSortField.CREATED_AT,
    order: SandboxListSortDirection.DESC,
  })) {
    console.log(sandbox.id)
  }
}

main().catch(console.error)
